static/js/controllers/contacts-content.js

```diff
@@ -211,29 +213,37 @@ var _ = require('underscore'),
         });
       };
 
-      var removePrimaryContact = function(children, primaryContactDoc) {
-        if (!children || children.length === 0 || !primaryContactDoc) {
-          return;
-        }
-        var primaryContactIdx = _.findIndex(children, function(child) {
-          return child.doc._id === primaryContactDoc._id;
-        });
-        if (primaryContactIdx < 0) {
-          return;
+      var placePrimaryContactOnTop = function(primaryContactDoc, persons) {
+        var removePrimaryContact = function(children, primaryContactDoc) {
+          if (!children || children.length === 0 || !primaryContactDoc) {
+            return;
+          }
+          var primaryContactIdx = _.findIndex(children, function(child) {
+            return child.doc._id === primaryContactDoc._id;
+          });
+          if (primaryContactIdx < 0) {
+            return;
+          }
+          children.splice(primaryContactIdx, 1);
+        };
+
+        if (primaryContactDoc) {
+          removePrimaryContact(persons, primaryContactDoc);
+          persons.unshift({ id: primaryContactDoc._id, doc: primaryContactDoc, isPrimaryContact: true });
         }
-        children.splice(primaryContactIdx, 1);
       };
 
-      var sortChildPersons = function(persons, contactDoc, primaryContactDoc) {
-        if (persons.length) {
+
+      var sortChildPersons = function(children, contactDoc, primaryContactDoc) {
```

Comment:
Yeah, I hear you. I prefer sending copies around too, I had that same thought at some point!
(don't think I don't see what you're doing, Mr. Functional Programming)

But there's a number of these array processing functions in the file, that are all doing the same kind of thing, and the sort functions are in-place (because using `Array.prototype.sort`) and so in the end the whole file became in-place modification instead of returning copies. In the end I opted for the same style everywhere, because if I do half-and-half THEN it's really sneaky, so in-place won.

I considered changing it all to copies though :) In the end I thought it wasn't worth it. Would you have? (interesting conversation)